src/test/java/org/opentripplanner/middleware/controllers/api/TrackedTripControllerTest.java

```diff
@@ -504,6 +516,111 @@ void canNotUpdateCompletedJourney() throws Exception {
+
+        // Reroute trip from 'deviated' position.
+        var reroutedItinerary = ManageTripTracking.rerouteTrip(
+            new TripTrackingData(rerouteMonitoredTrip, trackedJourney, List.of(deviatedPosition))
+        );
+        assertEquals(expectedReroutedItinerary.duration, reroutedItinerary.duration);
+        TrackedJourney updated = Persistence.trackedJourneys.getById(trackedJourney.id);
+        assertEquals(1, updated.reroutings.size());
+        assertEquals(trackedJourney.longestConsecutiveDeviatedPoints, updated.longestConsecutiveDeviatedPoints);
+
+        MonitoredTrip trip = Persistence.monitoredTrips.getById(rerouteMonitoredTrip.id);
+        assertEquals(expectedReroutedItinerary.duration, trip.journeyState.matchingItinerary.duration);
+        assertEquals(expectedReroutedItinerary.legs.size(), trip.journeyState.matchingItinerary.legs.size());
+
+        // Update tracking from start of rerouted position.
+        response = makeRequest(
+            UPDATE_TRACKING_TRIP_PATH,
+            JsonUtils.toJson(createUpdateTrackingPayload(
+                trackedJourney.id,
+                List.of(deviatedPosition)
+            )),
+            headers,
+            HttpMethod.POST
+        );
+
+        // Confirm traveler is no longer 'deviated'.
+        updateTrackingResponse = JsonUtils.getPOJOFromJSON(response.responseBody, TrackingResponse.class);
+        assertNotEquals(TripStatus.DEVIATED.name(), updateTrackingResponse.tripStatus);
+
+        // Reroute again from a different location.
+        trackedJourney.locations.clear();
+        trackedJourney.locations.add(reroutingPointPosition);
+        reroutedItinerary = ManageTripTracking.rerouteTrip(
+            new TripTrackingData(rerouteMonitoredTrip, trackedJourney, trackedJourney.locations)
+        );
+        assertEquals(expectedReroutedItinerary.duration, reroutedItinerary.duration);
+        updated = Persistence.trackedJourneys.getById(trackedJourney.id);
+        assertEquals(2, updated.reroutings.size());
+        assertEquals(trackedJourney.longestConsecutiveDeviatedPoints, updated.longestConsecutiveDeviatedPoints);
+        assertEquals(new Coordinates(updated.lastLocation()), reroutingPoint);
+
+        // Update tracking from start of the new rerouted position.
+        response = makeRequest(
+            UPDATE_TRACKING_TRIP_PATH,
+            JsonUtils.toJson(createUpdateTrackingPayload(
+                trackedJourney.id,
+                List.of(reroutingPointPosition)
+            )),
+            headers,
+            HttpMethod.POST
+        );
+
+        // Confirm traveler is still not 'deviated'.
+        updateTrackingResponse = JsonUtils.getPOJOFromJSON(response.responseBody, TrackingResponse.class);
+        assertNotEquals(TripStatus.DEVIATED.name(), updateTrackingResponse.tripStatus);
+    }
+
+    /** Provides a mock OTP 'plan' rerouted response. */
+    public Supplier<OtpResponse> mockOtpPlanResponse() {
```

Comment:
Rename to `mockOtpReroutedPlanResponse` or something similar.

Reply:
Updated in: https://github.com/ibi-group/otp-middleware/pull/286/commits/23d04728e3e4d229384d0e37b6ca821b9f416208